parquet-common/src/main/java/org/apache/parquet/bytes/CapacityByteArrayOutputStream.java

```diff
@@ -164,6 +164,15 @@ public CapacityByteArrayOutputStream(int initialSlabSize, int maxCapacityHint, B
   private void addSlab(int minimumSize) {
     int nextSlabSize;
 
+    // check for overflow 
+    try {
+      Math.addExact(bytesUsed, minimumSize);
+    } catch (ArithmeticException e) {
+      // This is interpreted as a request for a value greater than Integer.MAX_VALUE
+      // We throw OOM because that is what java.io.ByteArrayOutputStream also does
+      throw new OutOfMemoryError("Size of data exceeded 2GB (" + e.getMessage() + ")");
```

Comment:
The error message mismatches the check above. Should we compare with 2GB explicitly or change the message?

Reply:
The exception thrown by Math.exact will happen only if integer overflow occurs which happens when the data exceeds 2GB (i.e greater than `Integer.MAX_VALUE` ). 
What message were you thinking?

Reply:
If we simply do an overflow check here, then the error message should say `Integer.MAX_VALUE` instead of `2GB`. Otherwise, we should explicitly check if the addition result exceeds 2GB. WDYT?

Reply:
Let me just change the message.